espree.js

```diff
@@ -5114,7 +5113,9 @@ function parseClassDeclaration() {
 
     expectKeyword("class");
 
-    id = parseVariableIdentifier();
+    if (lookahead.type === Token.Identifier) {
```

Comment:
This isn't quite right because `parseClassDeclaration()` normally *must* have an identifier. So you really need to pass in a flag specifying whether the ID is optional or required.